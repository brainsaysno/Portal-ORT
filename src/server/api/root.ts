import { createTRPCRouter } from "./trpc";
import { materiaRouter } from "./routers/example";

/**
 * This is the primary router for your server.
 *
 * All routers added in /api/routers should be manually added here
 */
export const appRouter = createTRPCRouter({
  materia: materiaRouter,
});

// export type definition of API
export type AppRouter = typeof appRouter;
