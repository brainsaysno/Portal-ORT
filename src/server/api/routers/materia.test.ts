import { type inferProcedureInput } from "@trpc/server";
import { describe, expect, test } from "vitest";
import { type AppRouter, appRouter } from "../root";
import { createTRPCContext } from "../trpc";

describe("Materia router", () => {
  test("byId", async () => {
    const ctx = createTRPCContext();
    const caller = appRouter.createCaller(ctx);

    type Input = inferProcedureInput<AppRouter["materia"]["getById"]>;
    const input: Input = {
      id: 3851,
    };

    const arqSistemas = await caller.carrera.getById(input);

    expect(arqSistemas).toMatchSnapshot();
  });

  test("getPreviasById", async () => {
    const ctx = createTRPCContext();
    const caller = appRouter.createCaller(ctx);

    type Input = inferProcedureInput<AppRouter["materia"]["getPreviasById"]>;
    const input: Input = {
      id: 3851,
    };

    const arqSistemas = await caller.carrera.getById(input);

    expect(arqSistemas).toMatchSnapshot();
  });
});
