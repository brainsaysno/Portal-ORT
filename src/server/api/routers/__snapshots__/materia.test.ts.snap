// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Materia router > byId 1`] = `null`;

exports[`Materia router > getPreviasById 1`] = `null`;
